distributed/scheduler.py

```diff
@@ -8724,26 +8742,28 @@ def _materialize_graph(
     dsk2 = {}
     fut_deps = {}
     for k, v in dsk.items():
-        dsk2[k], futs = unpack_remotedata(v, byte_keys=True)
+        v, futs = unpack_remotedata(v, byte_keys=True)
         if futs:
             fut_deps[k] = futs
+
+        # Remove aliases {x: x}.
+        # FIXME: This is an artifact generated by unpack_remotedata when using persisted
+        # collections. There should be a better way to achieve that tasks are not self
+        # referencing themselves.
+        if not iskey(v) or v != k:
+            dsk2[k] = v
+
     dsk = dsk2
 
     # - Add in deps for any tasks that depend on futures
     for k, futures in fut_deps.items():
-        dependencies[k].update(f.key for f in futures)
+        dependencies[k].update(f.key for f in futures if f.key != k)
```

Comment:
This is from @fjetter and I'm unclear about the use case for it...

Rest of the changes to the function are non-functional.

Reply:
I'm going to revert it to unblock this PR. Once @fjetter responds, I'll potentially write a dedicated PR (with unit test to demonstrate the edge case).